pilot/cmd/pilot-agent/status/server.go

```diff
@@ -718,10 +718,12 @@ func (s *Server) handleAppProbeHTTPGet(w http.ResponseWriter, req *http.Request,
 		return
 	}
 
-	// Forward incoming headers to the application.
-	if prober.HTTPGet.Host != "" {
-		appReq.Host = prober.HTTPGet.Host
+	appReq.Host = req.Host
+	// trim port if any
+	if host, _, err := net.SplitHostPort(req.Host); err == nil {
+		appReq.Host = host
 	}
```

Comment:
At least for me it's ok if you strip the port because we are always setting the host header to a standard port, but in the general case I don't think this is ok - depends on the context in which this code is being used. It's definitely valid to put a port into a host header. I think you'd only want to do it if your ingress was on a non-standard port, but just putting it out there to consider.

I expect this value to be forwarded to the target app properly as `Host: example.com:5000`.

```yaml
httpHeaders:
  - name: Host
    value: 'example.com:5000'
```

Reply:
fixed